core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java

```diff
@@ -168,16 +170,20 @@ public long getSizeInBytes()
         return sizeInBytes;
     }
 
-    private void copyBytes(Block block, int[] lengths, int[] positions, int count, int[] targetOffsets, int targetOffsetsIndex, int newByteCount)
+    private void copyBytes(Slice rawSlice, int[] lengths, int[] sourceOffsets, int count, int newByteCount)
```

Comment:
As I mentioned in the description, part of the gain is from avoiding `getSlice` (~10% overall, 1/3 of the gain) and the other part is from using `System.arraycopy` instead of `Slice.getBytes`.
Theoretically allocating `Slice` in `getSlice` could be eliminated by inlining and escape analysis but it does not happen in practice, at least not in the jmh (the call is inlined but allocating is not eliminated).